javaparser-core/src/main/java/com/github/javaparser/ast/NodeList.java

```diff
@@ -161,4 +171,222 @@ public Node getParentNodeForChildren() {
         v.visit(this, arg);
     }
 
+    /**
+     * @param action
+     * @see java.lang.Iterable#forEach(java.util.function.Consumer)
+     */
+    @Override
+    public void forEach(Consumer<? super N> action) {
+        innerList.forEach(action);
+    }
+
+    /**
+     * @param o
+     * @return
+     * @see java.util.List#contains(java.lang.Object)
+     */
+    @Override
+    public boolean contains(Object o) {
+        return innerList.contains(o);
+    }
+
+    /**
+     * @return
+     * @see java.util.List#toArray()
+     */
+    @Override
+    public Object[] toArray() {
+        return innerList.toArray();
+    }
+
+    /**
+     * @param a
+     * @return
+     * @see java.util.List#toArray(java.lang.Object[])
+     */
+    @Override
+    public <T> T[] toArray(T[] a) {
+        return innerList.toArray(a);
+    }
+
+    /**
+     * @param o
+     * @return
+     * @see java.util.List#remove(java.lang.Object)
+     */
+    @Override
+    public boolean remove(Object o) {
+        return innerList.remove(o);
+    }
+
+    /**
+     * @param c
+     * @return
+     * @see java.util.List#containsAll(java.util.Collection)
+     */
+    @Override
+    public boolean containsAll(Collection<?> c) {
+        return innerList.containsAll(c);
+    }
+
+    /**
+     * @param c
+     * @return
+     * @see java.util.List#addAll(java.util.Collection)
+     */
+    @Override
+    public boolean addAll(Collection<? extends N> c) {
+        for (N n : c)
+            own(n);
+        return innerList.addAll(c);
+    }
+
+    /**
+     * @param index
+     * @param c
+     * @return
+     * @see java.util.List#addAll(int, java.util.Collection)
+     */
+    @Override
+    public boolean addAll(int index, Collection<? extends N> c) {
+        for (N n : c)
+            own(n);
+        return innerList.addAll(index, c);
+    }
+
+    /**
+     * @param c
+     * @return
+     * @see java.util.List#removeAll(java.util.Collection)
+     */
+    @Override
+    public boolean removeAll(Collection<?> c) {
+        return innerList.removeAll(c);
```

Comment:
Actually, I don't think we do that in other nodes as well, right? Hmmm, that means that the childNodes list will accumulate all kinds of crap when you start modifying the AST...